### ISSUE TITLE
libllvm does not create a native llvm-config

### ISSUE THREAD

**Issue by vishalbiswas** [termux/termux-packages#879]
Cross compiling llvm builds a native (host=build) version of `llvm-config` as a make step. But in termux, this is not the case. The native llvm-config is built for target host.
The native build takes place in `$TERMUX_PKG_BUILDDIR/NATIVE`. I was unable to find the root cause, but, the native build also uses `$TERMUX_STANDALONE_TOOLCHAIN/bin/clang++`. I suspect that this may be a bug in the cross compilation mechanism of cmake itself.

This is required to build rustc and other packages dependent on `llvm-config`.

**Comment by xeffyr** [comment]
Implemented in https://github.com/termux/termux-packages/pull/917.

**Comment by vishalbiswas** [comment]
This was kept open because the underlying bug which caused libllvm to not create a host=gnueabi/target=androideabi llvm-config was never fixed. It was just sidestepped.

**Comment by truboxl** [comment]
Relevant files to look at: https://github.com/llvm/llvm-project/tree/main/llvm/tools/llvm-config

I dive a little bit and I wonder if it has always been this way (`llvm-config` will be built for Target not Build)
Target = Android
Build = Linux

For our use case if we want to have cross `llvm-config` that can run on Build, maybe we should just host build manually ourselves.
Need to track down where the Android variables went to...

**Comment by thunder-coding** [comment]
Seems like this is no longer an issue, there is a native `llvm-config` binary in the latest libllvm package (I don't know when this got fixed). Closing this issue, kindly open a new issue if this is not the case since old issues are likely to raise less attention

**Comment by truboxl** [comment]
This is not about llvm-config in Termux.

This is about getting rid of https://github.com/termux/termux-packages/blob/10e774b21e9917f2b7e7617a065b7ab6de136474/scripts/build/termux_step_override_config_scripts.sh#L23 and https://github.com/termux/termux-packages/blob/master/packages/libllvm/llvm-config.in for cross compilation.

And have you checked with @finagolfin before closing?

**Comment by finagolfin** [comment]
Yeah, the issue is that LLVM does not properly compile a hostbuild llvm-config for linux x86_64, that we can then invoke when cross-compiling to Android.
